Makefile

```diff
@@ -1,19 +1,50 @@
-.PHONY: install build test lint vet fmt clean
+#
 
-install: fmt test
-	go install -v
+# Makefile for the downloader service.
+#
+# SYNOPSIS:
+#
+#   make             - format and compile the entire program along with its dependencies
+#   make fmt         - run gofmt to report any code formatting errors
+#   make deps        - verify consistency of `vendor/` directory contents
+#   make build       - compile the program but not install the results
+#   make check       - run the package's test suite
+#   make install     - compile the program and create the executables
+#   make clean       - remove all files generated by building the program
+#   make distclean   - remove all files generated by make
+#   make lint        - run golint linter on source code
+#   make vet         - run the Go vet command on source code
 
-test:
-	go test -race -p 1 ./...
+#
 
-lint:
-	golint ./...
+.PHONY: install fmt deps build check install clean distclean lint vet
 
-vet:
-	go vet ./...
+all: fmt build
 
 fmt:
 	! gofmt -d -e -s *.go **/*.go 2>&1 | tee /dev/tty | read
 
+deps:
+	dep ensure -v
+
+build: deps
+	go build -race ./...
+
+check: build
+	go test -race -p 1 ./...
+
+install: all
+	go install -v
+
 clean:
+	rm -rf vendor/
 	go clean
+
+distclean: clean
+	go clean -i -cache -testcache
+
+lint:
+	golint ./...
```

Comment:
Yeap, let's handle this on another PR.

Reply:
FWIW, we never used lint and I don't think we will (at least not in the short term).

Reply:
So, do you want me to completely remove it?

Reply:
IMHO linting should at least be advised ( since it is not currently forced ).
The project has 7 lint warnings, so it is surely not unlinted. In any case, i think we can have a correct lint target at least for whoever uses it.

Reply:
I also agree, let's keep this target as it is for now!

Reply:
TBH I expected the project to have many more lint warnings. All of them are valid too, so I agree, let's keep it and start using it.